lib/active_record/acts_as/relation.rb

```diff
@@ -28,7 +28,7 @@ def acts_as(name, scope = nil, options = {})
           validate :actable_must_be_valid if validates_actable
 
           before_save do
-            @_acting_as_changed = ActiveRecord.version.to_s.to_f >= 5.1 ? acting_as.saved_changes? : acting_as.changed?
+            @_acting_as_changed = ActiveRecord.version.to_s.to_f >= 5.1 ? acting_as.has_changes_to_save? : acting_as.changed?
```

Comment:
If I may be allowed to nitpick -- converting to string then float is dangerous. Gem::Version objects are comparable with each other, so this can be written as:

```ruby
if ActiveRecord.version >= Gem::Version.new('5.1')
```

The overhead for checking versions can be run once, so
```ruby
if ActiveRecord.version >= Gem::Version.new('5.1')
  before_save { @_acting_as_changed = acting_as.has_changes_to_save? }
else
  before_save { @_acting_as_changed = acting_as.changed? }
end
```

then we can extract this further:
```ruby
changed_method = ActiveRecord.version >= Gem::Version.new('5.1') ? :has_changes_to_save? : :changed?
before_save { @_acting_as_changed = send(changed_method) }
```